server/protocols/jmap/src/main/java/org/apache/james/jmap/model/GetMessageListRequest.java

```diff
@@ -115,13 +115,19 @@ public Builder fetchSearchSnippets(boolean fetchSearchSnippets) {
         }
 
         public GetMessageListRequest build() {
-            Preconditions.checkState(position >= 0, "'position' should be positive or null");
+            checkPosition();
             checkLimit();
             return new GetMessageListRequest(Optional.ofNullable(accountId), Optional.ofNullable(filter), sort.build(), Optional.ofNullable(collapseThreads),
-                    position, Optional.ofNullable(anchor), Optional.ofNullable(anchorOffset), Optional.ofNullable(limit), Optional.ofNullable(fetchThreads),
+                    Optional.ofNullable(position), Optional.ofNullable(anchor), Optional.ofNullable(anchorOffset), Optional.ofNullable(limit), Optional.ofNullable(fetchThreads),
                     Optional.ofNullable(fetchMessages), fetchMessageProperties.build(), Optional.ofNullable(fetchSearchSnippets));
         }
 
+        private void checkPosition() {
+            if (position != null) {
+                Preconditions.checkState(position >= 0, "'position' should be positive or null");
```

Comment:
We may want to check here we are not higher than 2^53

Reply:
We need some strong typing here...

It would be wonderful if we serialize / deserialize every number into a `Number` that handles these checks for us... There might be some Jackson checks

```
Number.fromInt(a);
Number.fromLong(b);
number.getValue();
```

@JsonValue will probably be useful.

We need to use Number for all numeric values server by JMAP:

 - SortOrder
 - Mailboxes numeric properties
 - Messages numeric properties
 - etc...

Reply:
I sill think we need strong typing for Number. Could you please introduce a JMAP Number class that handles the knowledge of what a JMAP spec `Number` is ?

Reply:
I don't think so, basically Integer and Long are extended from Number class and they had their limitation and I think it's enough, we do not need more Number class.

Reply:
No, I'm not speaking about the JDK Number class, I'm speaking about a specific **JMAP level** `Number` class, wrapping the definition of `what is a JMAP Number`. 

I don't see any downside argument about not doing it. I can help on this with some pair programming.